### DISCUSSION TITLE
TrackStacks.org

### DISCUSSION THREAD

**Discussion by varu1303** [scaleracademy/open-source-projects#25] [Ideas]
Only way to thrive in tech is by keep learning and there is no shortage of learning resources on internet. Problem 'TechStacks' aim to solve is 'What to learn?'

Any working professional with 1+ year of experience at least has 1 primary language that they work on daily basis. Let's take an example of JavaScript

JavaScript's ecosystem is constantly evolving and improving with time as dev community is always pushing for better performance and ease of development and deployment but working professionals have limited time to upskill and learn and are often at crossroads because of TOO MANY OPTIONS.

JS has Vue, React, Angular, Next.js, GraphQL, NodeJS, Deno, Jest, Jasmine, Mocha and many other frameworks and tools.

Idea is to take each Framework/Tool get the trend of how many times it is being searched on Google, Stack Overflow and being posted as part of Job descriptions on LinkedIn etc and let user easily compare trends. This will help people make better choices.

Add on:
We can build on top of this and let users login using their Github profiles which will let them
 - Choose language and frameworks they know
 - Choose things they want to learn after doing their research and allow them to link their git repositories to these so their progress and projects can be tracked
 - Have a Feed like Twitter where people can share what they have learnt or projects they have built or if any framework has helped them grow in their career, interview experiences around any framework, their experience with any framework or tool, if they solved a problem using any new tool/framework.
 
Users can choose from 3 levels of Proficiency in anything that they have added in their TechStack
 - Interested/Beginner | Has just started learning about it and exploring it
 - Intermediate | Know the basics and have created some Personal project
 - Advanced | Have worked and delivered multiple Enterprise projects
 
 If user's timeline tracked well and projects are arranged properly can even be used as a replacement for resumes while applying for jobs.

**Comment by championswimmer** [comment]
A good place for inspiration could be https://roadmap.sh btw

**Reply by varu1303** [reply]
> A good place for inspiration could be https://roadmap.sh btw

Yes, something like that but less intimidating and backed with data with a realistic timeline for each stage.
